proposals/v3/api.md

```diff
@@ -0,0 +1,1119 @@
+| *unit* | The unit of measure for the reading. If there is no unit, this will be `null`. |
+| *context* | A mapping of arbitrary values to provide additional context for the reading. |
+
+##### Error (500, 400)
+See: [Errors](#errors)
+
+
+---
+
+### Read Device
+```
+GET http://HOST:5000/synse/v3/read/{device_id}
+```
+
+Read from the specified device. This endpoint is effectively the same as using the [`/read`](#read)
+endpoint where the label matches the [device id tag](tags.md#auto-generated), e.g.
+`http://HOST:5000/synse/v3/read?tags=id:b33f7ac0`.
+
+#### URI Parameters
+
+| Parameter | Required | Description |
+| --------- | -------- | ----------- |
+| *device_id* | yes | The ID of the device to read. [Device IDs](ids.md) are globally unique. |
+
+
+#### Response
+
+##### OK (200)
+The response schema for the `/read/{device_id}` endpoint is the same as the response schema
+for the [`/read`](#read) endpoint.
+
+```json
+[
+  {
+    "device": "12bb12c1f86a86e",
+    "kind": "temperature",
+    "type": "temperature",
+    "value": 20.3,
+    "timestamp": "2018-02-01T13:47:40.395939895Z",
+    "unit": {
+      "symbol": "C",
+      "name": "degrees celsius"
+    },
+    "context": {
+      "host": "127.0.0.1",
+      "sample_rate": 8
+    }
+  }
+]
+```
+
+**Fields**
+
+| Field | Description |
+| ----- | ----------- |
+| *device* | The ID of the device which the reading originated from. |
+| *kind* | The device kind, as specified by the plugin. |
+| *type* | The device type, as specified by the plugin. This is the last element in the namespaced device kind. |
+| *value* | The value of the reading. |
+| *timestamp* | An RFC3339 timestamp describing the time at which the reading was taken. |
+| *unit* | The unit of measure for the reading. If there is no unit, this will be `null`. |
+| *context* | A mapping of arbitrary values to provide additional context for the reading. |
+
+##### Error (500, 404)
+See: [Errors](#errors)
+
+
+---
+
+### Read Cached
+```
+GET http://HOST:5000/synse/v3/readcached
+```
+
+Stream reading data from the registered plugins.
+
+All plugins have the capability of caching their readings locally in order to maintain
+a higher resolution of state beyond the poll frequency which Synse Server may request at.
+This is particularly useful for push-based plugins, where we would lose the pushed
+reading if it were not cached.
+
+At the plugin level, caching read data can be enabled, but is disabled by default. Even if
+disabled, this route will still return data for every device that supports reading on each
+of the configured plugins. When read caching is disabled, this will just return a dump of
+the current state for all readings which is maintained by the plugin.
+
+#### Query Parameters
+
+| Key | Description |
+| --- | ----------- |
+| start | An RFC3339 or RFC3339Nano formatted timestamp which specifies a starting bound on the cache data to return. If no timestamp is specified, there will not be a starting bound. |
```

Comment:
As far as I have been able to tell, the max precision of RFC3339 format is second, where RFC3339Nano is nanosecond. We don't need it all the way down to the nanosecond, but I think we'd probably want at least millisecond resolution.

Either way, I include them both here because the parser for RFC3339 and RFC3339Nano will work for both formats, e.g.
```
t1 (rfc3339):     2019-01-23T12:04:17-05:00
t2 (rfc3339nano): 2019-01-23T12:04:17.202792-05:00

--- parsing as rfc3339 ---
t1: 2019-01-23 12:04:17 -0500 EST
t2: 2019-01-23 12:04:17.202792 -0500 EST

--- parsing as rfc3339nano ---
t1: 2019-01-23 12:04:17 -0500 EST
t2: 2019-01-23 12:04:17.202792 -0500 EST
```

Reply:
standardized on RFC3339 (e4a4cb6)